R/java_unpack.R

```diff
@@ -0,0 +1,79 @@
+#' Unpack a Java distribution file into cache directory
+#' 
+#' @inheritParams java_install
+#' @inheritParams global_quiet_param
+#' @return A `character` vector containing of length 1 containing the path to the unpacked Java directory.
+#' @keywords internal
+java_unpack <- function(
+  java_distrib_path,
+  quiet = FALSE
+) {
+  platforms <- c("windows", "linux", "macos")
+  architectures <- c("x64", "aarch64", "arm64")
+  java_versions <- getOption("rJavaEnv.valid_major_java_versions")
+
+  # Extract information from the file name
+  filename <- basename(java_distrib_path)
+  parts <- strsplit(gsub("\\.tar\\.gz|\\.zip", "", filename), "-")[[1]]
+
+  # Guess the version, architecture, and platform
+  version <- parts[vapply(parts, function(x) x %in% java_versions, logical(1))][1]
+  arch <- parts[vapply(parts, function(x) x %in% architectures, logical(1))][1]
+  platform <- parts[vapply(parts, function(x) x %in% platforms, logical(1))][1]
```

Comment:
If I'm reading the code correctly, `parts` will be a character vector so you can use `%in%` directly.

```suggestion
    version <- parts[parts %in% java_versions][1]
    arch <- parts[parts %in% architectures][1]
    platform <- parts[parts %in% platforms][1]
```